test/junit/scala/collection/IndexedSeqTest.scala

```diff
@@ -12,7 +12,9 @@ import org.junit.runners.JUnit4
   * @tparam E the element type
   */
 @RunWith(classOf[JUnit4])
-abstract class IndexedTest[T, E] {
+abstract class IndexedTest[TT, EE] {
+  final type T = TT
+  final type E = EE
```

Comment:
Why?

Reply:
There's copy/paste in these tests (see below were impls of foldLeft/foldRight/iterator are identical). Exposing these types here mean that when we copy/paste we don't need to manually change the type signatures. There are 4 cases of this below but I was expecting a lot more :)